tools/deals/deals.go

```diff
@@ -0,0 +1,141 @@
+package deals
+
+import (
+	"context"
+	"strings"
+
+	"github.com/zondax/golem/pkg/logger"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/manifest"
+	filTypes "github.com/filecoin-project/lotus/chain/types"
+	"github.com/zondax/fil-parser/metrics"
+	"github.com/zondax/fil-parser/parser"
+	"github.com/zondax/fil-parser/parser/helper"
+	"github.com/zondax/fil-parser/types"
+)
+
+const (
+	txStatusOk = "ok"
+)
+
+type EventGenerator interface {
+	GenerateDealsEvents(ctx context.Context, transactions []*types.Transaction, tipsetCid string, tipsetKey filTypes.TipSetKey) (*types.DealsEvents, error)
+}
+
+var _ EventGenerator = &eventGenerator{}
+
+type eventGenerator struct {
+	helper  *helper.Helper
+	logger  *logger.Logger
+	metrics *dealsMetricsClient
+	network string
+}
+
+func NewEventGenerator(helper *helper.Helper, logger *logger.Logger, metrics metrics.MetricsClient, network string) EventGenerator {
+	return &eventGenerator{
+		helper:  helper,
+		logger:  logger,
+		metrics: newClient(metrics, "deals"),
+		network: network,
+	}
+}
+
+func (eg *eventGenerator) GenerateDealsEvents(ctx context.Context, transactions []*types.Transaction, tipsetCid string, tipsetKey filTypes.TipSetKey) (*types.DealsEvents, error) {
+	events := &types.DealsEvents{
+		DealsMessages:    []*types.DealsMessages{},
+		DealsProposals:   []*types.DealsProposals{},
+		DealsActivations: []*types.DealsActivations{},
+		DealsSpaceInfo:   []*types.DealsSpaceInfo{},
```

Comment:
These are going to be 4 tables? wow! I wonder how the state is going to be reconstructed!